### DISCUSSION TITLE
String enum resolution?

### DISCUSSION THREAD

**Discussion by mary-ext** [bluesky-social/atproto#2389] [Clarifications and "Why?"]
This definition seems to be the only one that makes use of this functionality, where if the value starts with `#` then the resulting types would start with `lex:` with the full NSID written, suggesting it's being turned into references?

https://github.com/bluesky-social/atproto/blob/6b3addd73333c18675d1a35be0589438f5a2ee7e/lexicons/tools/ozone/moderation/defs.json#L165-L173

https://github.com/bluesky-social/atproto/blob/6b3addd73333c18675d1a35be0589438f5a2ee7e/packages/api/src/client/types/tools/ozone/moderation/defs.ts#L133-L138

Meanwhile, on another enum...

https://github.com/bluesky-social/atproto/blob/6b3addd73333c18675d1a35be0589438f5a2ee7e/lexicons/com/atproto/moderation/defs.json#L5-L16

https://github.com/bluesky-social/atproto/blob/6b3addd73333c18675d1a35be0589438f5a2ee7e/packages/api/src/client/types/com/atproto/moderation/defs.ts#L9-L17

I'm not sure if anything's been documented in this area, and I've just been leaving it alone in my own code generation tools because I'm not quite sure what to do with it.

**Comment by bnewbold** [comment]
cc: @devinivy we discussed this briefly this morning

I'm not really sure what is going on here in the TypeScript code. My understanding/assumption is that these are not really even refs, just strings. Probably we should update the lexicon itself to have the fully-qualified references, not just the fragment. I'm particularly confused by the "lex:" prefix, have not seen that anywhere else.
